### PR TITLE
Update dependencies and remove automated plugins for other projects

### PR DESCRIPTION
**Pull request by shumelchyk**
See https://github.com/android/wear-os-samples/issues/213 for context.

### CODE REVIEW

**Comment by shumelchyk** [comment]
@yschimke if you're online it may be good to do an interim review before I'll update the other projects.

File: AlwaysOnKotlin/gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
-compose-ui = "1.1.0-rc01"
 androidx-test = "1.4.0"
-androidx-wear-compose = "1.0.0-alpha13"
+androidx-wear-compose = "1.0.0-alpha16"
+compose = "1.2.0-alpha03"
 org-jetbrains-kotlinx = "1.6.0"
 
 [libraries]
+android-build-gradle = "com.android.tools.build:gradle:7.1.1"
+android-lint-gradle = "com.android.tools.lint:lint-gradle:30.1.1"
```
Comment (yschimke) [inline AlwaysOnKotlin/gradle/libs.versions.toml:10]: This seems stricter, from CI failures.

```
> Task :views:lintDebug FAILED
/home/runner/work/wear-os-samples/wear-os-samples/AlwaysOnKotlin/views/src/main/AndroidManifest.xml:43: Error: Redundant label can be removed [RedundantLabel]
            android:label="@string/app_name"
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: RuntimePermissionsWear/gradle/libs.versions.toml [hunk 2]
```diff
@@ -3,73 +3,73 @@ org-jetbrains-kotlin = "1.6.0"
 org-jetbrains-kotlinx = "1.6.0"
 
 [libraries]
+android-build-gradle = "com.android.tools.build:gradle:7.1.1"
+android-lint-gradle = "com.android.tools.lint:lint-gradle:30.1.1"
```
Comment (yschimke) [inline RuntimePermissionsWear/gradle/libs.versions.toml:7]: Same 

```
/home/runner/work/wear-os-samples/wear-os-samples/RuntimePermissionsWear/Application/src/main/AndroidManifest.xml:38: Error: Redundant label can be removed [RedundantLabel]
            android:label="@string/app_name"
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: WatchFaceKotlin/build.gradle [hunk 3]
```diff
@@ -42,31 +24,21 @@ buildscript {
...
+            targetExclude("$buildDir/**/*.kt")
+            targetExclude("bin/**/*.kt")
 
-tasks.named("dependencyUpdates").configure {
-    rejectVersionIf {
-        (it.candidate.version ==~ /.*-alpha.*/ && !(it.currentVersion ==~ /.*-alpha.*/)) ||
-                (it.candidate.version ==~ /.*-beta.*/ && !(it.currentVersion ==~ /.*-(beta|alpha).*/)) ||
-                (it.candidate.version ==~ /.*1.6.20-M.*/)
+            ktlint("0.41.0")
+            licenseHeaderFile rootProject.file("spotless/copyright.kt")
```
Comment (yschimke) [inline WatchFaceKotlin/build.gradle:41]: Missing, but worth adding if it doesn't trigger a bunch of failures.

```
> java.nio.file.NoSuchFileException: /home/runner/work/wear-os-samples/wear-os-samples/WatchFaceKotlin/spotless/copyright.kt
```

**Review by yschimke — APPROVED** [review]
After the build errors.  Thanks for following up on this so quickly.

File: AlwaysOnKotlin/gradle/libs.versions.toml (hunk 1 above)
Comment (shumelchyk) [reply AlwaysOnKotlin/gradle/libs.versions.toml:10]: Yeah, but it's even better. Should be fixed now

**Comment by yschimke** [comment]
LGTM

File: WatchFaceKotlin/build.gradle (hunk 3 above)
Comment (shumelchyk) [reply WatchFaceKotlin/build.gradle:42]: Copyright was missing (fixed now), but ktlint check fails as well, commented out it temporarily, but plan to take a look a bit later.

**Comment by yschimke** [comment]
At least some of the android test failures seem unrelated, just timeouts.

**Comment by shumelchyk** [comment]
@yschimke yeah seem so, I checked all of them locally before pushing and they run successfully.

**Comment by yschimke** [comment]
I know it was a bit special this time. But hopefully we can make it simpler/automated and routine in future.

**Comment by yschimke** [comment]
Failures seem unrelated

```
 com.google.testing.platform.core.device.DeviceProviderException: ErrorName: DEVICE_PROVISION_FAILED
NameSpace: com.google.testing.platform
ErrorCode: 3002
```

and

```

> Task :compose:connectedDebugAndroidTest

com.example.android.wearable.wear.alwayson.AlwaysOnAppTests > textIsCorrectAfterReturningToActiveMode[test(AVD) - 11] FAILED 
	java.lang.IllegalStateException: No compose views found in the app. Is your Activity resumed?
	at androidx.compose.ui.test.TestContext.getAllSemanticsNodes$ui_test_release(TestOwner.kt:96)
```

**Comment by shumelchyk** [comment]
Merging this PR with a note to investigate a possible workaround for flaky tests that seems to be a root cause for check failures in `AlwaysOnKotlin` sample.

**Comment by yschimke** [comment]
Raised https://github.com/android/wear-os-samples/issues/219

**Pull request merged**